### PR TITLE
Improve sharable feature.

### PR DESCRIPTION
**Pull request by SwithFr**
Simple sharing on social media needs some open graph meta.

- Add sharable properties for news CRUD (title, image and description)
- Possibility to disable properties with config.

### CODE REVIEW

**Comment by Meroje** [comment]
You don't need new fields, use the content nugget https://github.com/novius-os/noviusos_blognews/blob/c0c29d9/classes/model/post.model.php#L159-L198

**Comment by SwithFr** [comment]
Removed new fields and use content nugget instead.

**Review by Meroje — APPROVED** [review]

File: classes/controller/front.ctrl.php [hunk 1]
```diff
@@ -432,6 +435,31 @@ public function display_item()
...
+
+            $title = isset($sharableData['title']) ? $sharableData['title'] : $post->post_title;
+            $this->main_controller->addMeta('<meta property="og:title" content="' . $title . '" />');
+
+            $description = isset($sharableData['text']) ? $sharableData['text'] : $post->post_summary;
+            $this->main_controller->addMeta('<meta property="og:description" content="' . $description . '" />');
+
+            $this->main_controller->addMeta('<meta property="og:url" content="' . $post->url() . '" />');
+
+            $img_id = isset($sharableData['image']) ? (int)$sharableData['image'] : isset($post->medias->thumbnail) ? $post->medias->thumbnail->url() : null;
```
Comment (jguyomard) [inline classes/controller/front.ctrl.php]: camelCase?

File: config/controller/admin/post.config.php [hunk 2]
```diff
@@ -186,9 +185,9 @@ function ($item, $data) {
                     if (!empty($data['categories'])) {
                         $class = get_class($item);
                         $namespace = substr($class, 0, strrpos($class, '\\'));
-                        $category_class = $namespace.'\\Model_Category';
+                        $category_class = $namespace . '\\Model_Category';
```
Comment (jguyomard) [inline config/controller/admin/post.config.php]: please use concat_without_space style

File: config/controller/admin/post.config.php [hunk 3]
```diff
@@ -186,9 +185,9 @@ function ($item, $data) {
...
 
-                        $item->categories = $category_class::find('all', array('where' => array(array('cat_id', 'IN', (array) $data['categories']))));
+                        $item->categories = $category_class::find('all', array('where' => array(array('cat_id', 'IN', (array)$data['categories']))));
```
Comment (jguyomard) [inline config/controller/admin/post.config.php]: keep space after (array)

File: classes/controller/front.ctrl.php [hunk 4]
```diff
@@ -432,6 +435,31 @@ public function display_item()
...
         }
 
+        if (\Arr::get($this->app_config, 'share_properties.enabled')) {
+            $sharableData = $post->get_catcher_nuggets(Model_Content_Nuggets::DEFAULT_CATCHER)->content_data;
+            $this->main_controller->addMeta(
+                '<meta property="og:type" content="article" />'
+            );
+
+            $title = isset($sharableData['title']) ? $sharableData['title'] : $post->post_title;
+            $this->main_controller->addMeta('<meta property="og:title" content="' . $title . '" />');
```
Comment (jguyomard) [inline classes/controller/front.ctrl.php]: please use concat_without_space style

**Review by jguyomard — CHANGES_REQUESTED** [review]

**Review by jguyomard — APPROVED** [review]

**Pull request merged**